api/src/Messages/State/NumberOfMessageNotReadForCustomerProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Messages\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use App\Entity\User;
+use App\Messages\ApiResource\MessageUnread;
+use App\Repository\DiscussionMessageRepository;
+use Symfony\Bundle\SecurityBundle\Security;
+
+/**
+ * @template-implements ProviderInterface<MessageUnread>
+ */
+readonly class NumberOfMessageNotReadForCustomerProvider implements ProviderInterface
+{
+    public function __construct(
+        private DiscussionMessageRepository $discussionMessageRepository,
+        private Security $security,
+    ) {
+    }
+
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        /** @var ?User $user */
+        $user = $this->security->getUser();
```

Comment:
Don't forget to check instanceof $user, for example
```
if (!$user instanceof User) {
     throw new UnexpectedTypeException($user, User::class);
}
```

then 
```
$notRead = $$this->discussionMessageRepository->getNumberOfMessageNotReadForCustomer(user: $user);
```

because if user is null, there was an issue somewhere, so don't return any value

Reply:
Quoique, je me pose la question sur l'exception à retourner pas sur du UnexpectedTypeException

Reply:
Yes, I was thinking about what is the best to do if in a strange scenario :imp: where the user is null, if we have to return an empty result like you do with a response and the `$notRead` set to zero, or if throwing an exception will be better?

Or perhaps the good way in this case is to return the response with zero like you do but use a logger in the case of user is null, for example: 
```
 if (!user instanceof User) {
      $this->logger->error('User not found trying to get unread messages for customer');
}
```

Reply:
After some investigations, I think your first proposal is the best.